File: src/main/org/nlogo/api/ImporterUser.java
package org.nlogo.api;

import java.util.List;

public interface ImporterUser 
	extends WorldResizer
{ 
    // handle output area
    void setOutputAreaContents( String text ) ;
	// handle importing plots
	void currentPlot( String plot )  ;
	PlotInterface getPlot( String plot ) ;
	// handle importing extensions
	boolean isExtensionName( String name ) ;
	void importExtensionData( String name , List<String[]> data , 
							  ImportErrorHandler handler ) 
		throws ExtensionException ;
}

File: headless/src/main/org/nlogo/api/ImporterUser.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.api

import java.util.{ List => JList }

trait ImporterUser extends WorldResizer {
  def setOutputAreaContents(text: String)
  def currentPlot(plot: String)  // for importing plots
  def getPlot(plot: String): PlotInterface
  def isExtensionName(name: String): Boolean  // for importing extensions
  @throws(classOf[ExtensionException])
  def importExtensionData(name: String, data: JList[Array[String]], handler: ImportErrorHandler)
}
